core/ledger/kvledger/snapshot_test.go

```diff
@@ -99,15 +117,42 @@ func TestGenerateSnapshot(t *testing.T) {
 	require.NoError(t, kvlgr.CommitLegacy(blockAndPvtdata3, &ledger.CommitOptions{}))
 	require.NoError(t, kvlgr.generateSnapshot())
 	verifySnapshotOutput(t,
-		snapshotRootDir,
-		kvlgr.ledgerID,
-		4,
-		protoutil.BlockHeaderHash(blockAndPvtdata3.Block.Header),
-		kvlgr.commitHash,
-		"txids.data", "txids.metadata",
-		"public_state.data", "public_state.metadata",
-		"private_state_hashes.data", "private_state_hashes.metadata",
-		"confighistory.data", "confighistory.metadata",
+		&expectedSnapshotOutput{
+			snapshotRootDir:   snapshotRootDir,
+			ledgerID:          kvlgr.ledgerID,
+			ledgerHeight:      4,
+			lastBlockHash:     protoutil.BlockHeaderHash(blockAndPvtdata3.Block.Header),
+			previousBlockHash: blockAndPvtdata3.Block.Header.PreviousHash,
+			lastCommitHash:    kvlgr.commitHash,
+			stateDBType:       simpleKeyValueDB,
+			expectedBinaryFiles: []string{
+				"txids.data", "txids.metadata",
+				"public_state.data", "public_state.metadata",
+				"private_state_hashes.data", "private_state_hashes.metadata",
+				"confighistory.data", "confighistory.metadata",
+			},
+		},
+	)
+}
+
+func TestSnapshotDBTypeCouchDB(t *testing.T) {
+	conf, cleanup := testConfig(t)
+	defer cleanup()
+	provider := testutilNewProvider(conf, t, &mock.DeployedChaincodeInfoProvider{})
+	defer provider.Close()
+	lgr, err := provider.open("testLedger")
+	require.NoError(t, err)
+	kvlgr := lgr.(*kvLedger)
+
+	// artificially set the db type
```

Comment:
Maybe we should move the snapshot test to `kvledger/tests` so that we can run the test with both `GoLevelDB` and `CouchDB`. If we agree to do this, it can be done in a separate PR.

Reply:
Yes, I believe that we should retire all the tests in the `kvledger` package (they are just some form of duplicate) and have tests only at one place. I plan to move this into `kvledger/tests` when add function for bootstrap at this level. BTW, Ideally, sometime we should  think of moving `kvledger/tests` into `kvledger` folder but specify in `kvledger_test` that way they will serve both the purposes of external API testing and counting towards coverage.

Reply:
Agree.